tests/range_detection.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+import numpy as np
+import pandas as pd
+
+from anomalydetection.detectors import RangeDetector, DiffRangeDetector, AnomalyDetectionPipeline
+
+
+@pytest.fixture
+def range_data():
+    normal_data = np.array([0, np.nan, 1, 0, 2, np.nan, 3.14, 4])
+    abnormal_data = np.array([-1, np.nan, 2, np.nan, 1, 0, 4, 10])
+    expected_anomalies = np.array([True, False, False, False, False, False, True, True])
+    assert len(expected_anomalies) == len(abnormal_data)
+    return normal_data, abnormal_data, expected_anomalies
+
+
+@pytest.fixture
+def range_data_series(range_data):
+    normal_data, abnormal_data, expected_anomalies = range_data
+    time = pd.date_range(start='2020', periods=len(normal_data), freq='1H')
+    return pd.Series(normal_data, index=time), pd.Series(abnormal_data, index=time), expected_anomalies
+
+
+def test_range_detector(range_data_series):
+    data, _, _ = range_data_series
+
+    detector = RangeDetector(0, 2)
+    anomalies = detector.detect(data)
+
+    assert len(anomalies) == len(data)
+    assert sum(anomalies) == 2
+
+
+def test_range_detector_autoset(range_data_series):
```

Comment:
Similar comments as above:
Paranthesis needed to call range_data_series

I would prefer exact tests that assert that detected anomalies are equal to those expected